src/main/java/dbaccessor/user/UserAccessor.java

```diff
@@ -41,18 +42,34 @@ public void insertGivenUserId(User user) {
         spannerTemplate.insert(user);
     }
 
-    /* Checks if a user with the given username or email-id already exists in User table */
-    public boolean checkIfUserExists(String username, String emailID) {
-        String SQLStatment = "SELECT Username FROM User WHERE Username=@Username OR EmailID=@EmailID";
+    /**
+     * Checks if there exists users having given username or email-id
+     * If no such user exists, return 0
+     * If a user having only the given username exists return 1
+     * If a user having only the given email-id exists return 2
+     * If a user having both given username and email-id exists return 3
+     */
+    public int checkIfUsernameOrEmailIdExists(String username, String emailID) {
+        String SQLStatment = "SELECT Username, EmailID FROM User WHERE Username=@Username OR EmailID=@EmailID";
         Statement statement = Statement.newBuilder(SQLStatment)
                                 .bind("Username")
                                 .to(username)
                                 .bind("EmailID")
                                 .to(emailID)
                                 .build();
-        return !spannerTemplate
-                .query(User.class, statement, new SpannerQueryOptions().setAllowPartialRead(true)) //setAllowPartialRead for reading specific columns
-                .isEmpty();
+        List<User> resultSet = spannerTemplate.query(User.class, statement, new SpannerQueryOptions().setAllowPartialRead(true));
+        int usernameBit = 0;
+        int emailIdBit = 1;
+        int setBits = 0;
+        for(User user: resultSet) {
+            if(user.getUsername().equals(username)) {
+                setBits |= (1 << usernameBit);
+            }
+            if(user.getEmailID().equals(emailID)) {
+                setBits |= (1 << emailIdBit);
+            }
+        }
```

Comment:
This though is efficient space wise it is hard to read and error prone.

Consider making an enum and return an `EnumSet` instead for errors.
Small enum sets are implemented as bit sets so the efficiency is maintained while the code becomes much more readable.

Reply:
Done.